install/detect-platform.sh

```diff
@@ -0,0 +1,30 @@
+echo "${_group}Detecting Docker platform"
+
+
+# Sentry SaaS uses stock Yandex ClickHouse, but they don't provide images that
+# support ARM, which is relevant especially for Apple M1 laptops, Sentry's
+# standard developer environment. As a workaround, we use an altinity image
+# targeting ARM.
+#
+# See https://github.com/getsentry/self-hosted/issues/1385#issuecomment-1101824274
+#
+# Images built on ARM also need to be tagged to use linux/arm64 on Apple
+# silicon Macs to work around an issue where they are built for
+# linux/amd64 by default due to virtualization.
+# See https://github.com/docker/cli/issues/3286 for the Docker bug.
+
+DOCKER_ARCH=$(docker info --format '{{.Architecture}}')
+
+if [[ "$DOCKER_ARCH" = "x86_64" ]]; then
+    export DOCKER_PLATFORM="linux/amd64"
+    export CLICKHOUSE_IMAGE="yandex/clickhouse-server:20.3.9.70"
+elif [[ "$DOCKER_ARCH" = "aarch64" ]]; then
+    export DOCKER_PLATFORM="linux/arm64"
+    export CLICKHOUSE_IMAGE="altinity/clickhouse-server:20.10.1.4844-testing-arm"
```

Comment:
Do you think it's worth trying to keep this in sync with [our dev envs](https://github.com/getsentry/sentry/blob/2619db88ab9eacfdda77bb362813ce20536a5667/src/sentry/conf/server.py#L1931-L1935)?

At least with comments in both places. Also this isn't _quite_ in sync off the bat. 🙈 I picked this version because it's the closes to 20.3.9.70 I could find. Maybe decide with Anthony?

Reply:
Yeah I think that is probably a good idea, it at least means any issues can be tested with the dev environment. I'll discuss it with Anthony.